### PR TITLE
Added id to the interface

### PR DESCRIPTION
**Pull request by Buckram123**
Fixes ORC-56

As discussed in a ticket if interface contains id at the end we auto-generate `new` constructor with this "contract_id" or make users build their own constructor.

Currently supported syntax:
```rust
#[interface(InstantiateMsg, ExecuteMsg, QueryMsg, MigrateMsg, id=MY_CONTRACT)]
// or 
#[interface(InstantiateMsg, ExecuteMsg, QueryMsg, MigrateMsg, id="contract_literal")]
```
It is up for discussion if we want to change it or add more restrictions

Closes https://github.com/AbstractSDK/cw-orchestrator/issues/96

TODO:
- [ ] Update docs and comments according to the changes (after we agree on syntax)

### CODE REVIEW

**Comment by Buckram123** [comment]
> I'd prefer using the const as the ID is something that's often used elsewhere

Right now it supports both or would you prefer for it to be only const?

File: packages/cw-orch-contract-derive/src/lib.rs [hunk 1]
```diff
@@ -163,19 +192,26 @@ pub fn interface(attrs: TokenStream, input: TokenStream) -> TokenStream {
     };
 
     let name = cw_orch_struct.ident.clone();
+    let default_num = if let Some(id_expr) = default_id {
```
Comment (Kayanski) [inline packages/cw-orch-contract-derive/src/lib.rs:195]: I don't like setting the id forcefully here. Can we keep the old version if no id was provided and put this new version only if a default id is provided ?

File: packages/cw-orch-contract-derive/src/lib.rs [hunk 2]
```diff
@@ -163,19 +192,26 @@ pub fn interface(attrs: TokenStream, input: TokenStream) -> TokenStream {
...
+            impl <Chain: ::cw_orch::prelude::CwEnv, #all_generics> #name<Chain, #all_generics> {
+                pub fn new(chain: Chain) -> Self {
+                    Self(
+                        ::cw_orch::contract::Contract::new(#id_expr, chain)
+                    , #(#all_phantom_marker_values,)*)
+                }
+            }
+        )
+    } else {
+        quote!()
```
Comment (Kayanski) [inline packages/cw-orch-contract-derive/src/lib.rs:206]: Like simply, reput the old version here

File: packages/cw-orch-contract-derive/src/lib.rs (hunk 1 above)
Comment (Buckram123) [reply packages/cw-orch-contract-derive/src/lib.rs:195]: That's actually the goal to make it not generate `new` by default so it's handled by the contract maintainer

Comment (Kayanski) [reply packages/cw-orch-contract-derive/src/lib.rs:195]: Oh, hmm, ok ! Adds an extra argument that has nothing to do with messages, but that could work

Comment (Buckram123) [reply packages/cw-orch-contract-derive/src/lib.rs:195]: > Oh, hmm, ok ! Adds an extra argument that has nothing to do with messages, but that could work

We totally could change syntax of it like 
```rust
#[interface(InstantiateMsg, ExecuteMsg, QueryMsg, MigrateMsg)]
#[contract_id=CONTRACT_ID]
```

**Pull request merged**